services/src/datasets/external/netcdfcf/overviews.rs

```diff
@@ -0,0 +1,571 @@
+    start: TimeInstance,
+    end: TimeInstance,
+    step: TimeStep,
+}
+
+fn store_metadata(provider_path: &Path, dataset_path: &Path, out_folder_path: &Path) -> Result<()> {
+    let out_file_path = out_folder_path.join(METADATA_FILE_NAME);
+
+    if out_file_path.exists() {
+        debug!("Skipping metadata generation: {}", dataset_path.display());
+        return Ok(());
+    }
+
+    debug!("Creating metadata: {}", dataset_path.display());
+
+    let metadata = NetCdfCfDataProvider::build_netcdf_tree(provider_path, None, dataset_path)?;
+
+    fs::create_dir_all(out_folder_path).boxed_context(error::CannotCreateOverviews)?;
+
+    let file = File::create(out_file_path).boxed_context(error::CannotWriteMetadataFile)?;
+
+    let mut writer = BufWriter::new(file);
+
+    writer
+        .write_all(
+            serde_json::to_string(&metadata)
+                .boxed_context(error::CannotWriteMetadataFile)?
+                .as_bytes(),
+        )
+        .boxed_context(error::CannotWriteMetadataFile)?;
+
+    Ok(())
+}
+
+fn index_subdataset(conversion: &ConversionMetadata, time_coverage: &TimeCoverage) -> Result<()> {
+    const COG_BLOCK_SIZE: &str = "512";
+    const COMPRESSION_FORMAT: &str = "LZW";
+    const COMPRESSION_LEVEL: &str = "9";
+
+    if conversion.dataset_out.exists() {
+        debug!("Skipping conversion: {}", conversion.dataset_out.display());
+        return Ok(());
+    }
+
+    debug!("Indexing conversion: {}", conversion.dataset_out.display());
+
+    let subdataset = gdal_open_dataset_ex(
+        Path::new(&conversion.dataset_in),
+        DatasetOptions {
+            open_flags: GdalOpenFlags::GDAL_OF_READONLY,
+            allowed_drivers: Some(&["netCDF"]),
+            open_options: None,
+            sibling_files: None,
+        },
+    )
+    .boxed_context(error::CannotOpenNetCdfSubdataset)?;
+
+    fs::create_dir_all(
+        conversion
+            .dataset_out
+            .parent()
+            .context(error::CannotReadInputFileDir)?,
+    )
+    .boxed_context(error::CannotCreateOverviews)?;
+
+    let gdal_num_threads = get_config_element::<crate::util::config::Gdal>()
+        .boxed_context(error::CannotCreateOverviews)?
+        .compression_num_threads;
+    let num_threads = gdal_num_threads.to_string();
+
+    let cog_driver = gdal::Driver::get("COG").boxed_context(error::CannotCreateOverviews)?;
+    let options = vec![
+        RasterCreationOption {
+            key: "COMPRESS",
+            value: COMPRESSION_FORMAT,
+        },
+        RasterCreationOption {
+            key: "LEVEL",
+            value: COMPRESSION_LEVEL,
+        },
+        RasterCreationOption {
+            key: "NUM_THREADS",
+            value: &num_threads,
+        },
+        RasterCreationOption {
+            key: "BLOCKSIZE",
+            value: COG_BLOCK_SIZE,
+        },
+        RasterCreationOption {
+            key: "BIGTIFF",
+            value: "IF_SAFER", // TODO: test if this suffices
```

Comment:
suffices for what?

Reply:
It could be, that this option suffices for all data that we have and will get. If not, we have to create a manual switch.